roles/StackStorm.st2/tasks/datastore.yml

```diff
@@ -13,6 +13,14 @@
   command: st2-generate-symmetric-crypto-key --key-path {{ st2_datastore_key_file }}
   args:
     creates: "{{ st2_datastore_key_file }}"
+  when: not st2_datastore_key
+
+- name: Copy st2 encryption key
+  become: yes
+  copy:
+    content: "{{ st2_datastore_key }}"
+    dest: "{{ st2_datastore_key_file }}"
```

Comment:
When the content of the file changes we'll need to notify st2 to restart the services.
According to code below it's `st2api`, but I guess that's incomplete as more StackStorm services might be using the crypto key.

Reply:
@armab added a restart to all of the st2 services

Reply:
Restarting all the st2 services is impractical here. We need only specific services that are using st2 datastore key.
Can you please figure out which may use it?